site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java

```diff
@@ -0,0 +1,125 @@
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ ******************************************************************************/
+package org.sakaiproject.groupmanager.service;
+
+import java.util.Locale;
+
+import javax.inject.Inject;
+
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.event.api.EventTrackingService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.site.api.Group;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.ToolManager;
+import org.sakaiproject.user.api.PreferencesService;
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class SakaiService  {
+
+    @Inject
+    private EventTrackingService eventTrackingService;
+
+    @Inject
+    private SiteService siteService;
+
+    @Inject
+    private PreferencesService preferencesService;
+
+    @Inject
+    private ServerConfigurationService serverConfigurationService;
+
+    @Inject
+    private SessionManager sessionManager;
+
+    @Inject
+    private ToolManager toolManager;
+
+    @Inject
+    private UserDirectoryService userDirectoryService;
+
+    public Site getCurrentSite() {
+        String siteId = toolManager.getCurrentPlacement().getContext();
+        Site site = null;
+        try {
+            site = siteService.getSite(siteId);
+        } catch (Exception ex) {
+            log.error("Unable to find the site with Id {}.", siteId);
+            return null;
+        }
+        return site;
+    }
+
+    public Locale getCurrentUserLocale() {
+        String userId = sessionManager.getCurrentSessionUserId();
+        return preferencesService.getLocale(userId);
+    }
+
+    public String getCurrentUserId() {
+        return sessionManager.getCurrentSessionUserId();
+    }
+
+    public boolean saveSite(Site site) {
+        try {
+            siteService.save(site);
+        } catch (IdUnusedException | PermissionException e) {
+            log.error("Error saving the site {}.", site.getId());
+            return false;
+        }
+        return true;
+    }
+
+    public User getUser(String userId) {
+        try {
+            return userDirectoryService.getUser(userId);
+        } catch (UserNotDefinedException e) {
+            log.error("Unable to get user by id {}.", userId);
+        }
+        return null;
```

Comment:
I would personally return an Optional from this.

Reply:
sure, good pattern.